From fcb4dd0c831339c4374ace17d8f2ae6ebfeed279 Mon Sep 17 00:00:00 2001
From: "Ondrej Zajicek (work)" <santiago@crfreenet.org>
Date: Sun, 27 Feb 2022 03:10:38 +0100
Subject: [PATCH] Babel: Fix bug in iface reconfiguration

A recent change in Babel causes ifaces to disappear after
reconfiguration. The patch fixes that.

Thanks to Johannes Kimmel for an insightful bugreport.
---
 proto/babel/babel.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/proto/babel/babel.c b/proto/babel/babel.c
index e43818f5..174fc9e2 100644
--- a/proto/babel/babel.c
+++ b/proto/babel/babel.c
@@ -1899,7 +1899,7 @@ babel_reconfigure_ifaces(struct babel_proto *p, struct babel_config *cf)
     struct babel_iface *ifa = babel_find_iface(p, iface);
     struct babel_iface_config *ic = (void *) iface_patt_find(&cf->iface_list, iface, NULL);
 
-    if (ic && iface_is_valid(p, iface))
+    if (ic && !iface_is_valid(p, iface))
       ic = NULL;
 
     if (ifa && ic)
-- 
2.35.1

